### ISSUE TITLE
Add automatic convergence-checks

### ISSUE THREAD

**Issue by AdamCSmithCWS** [BrandonEdwards/bbsBayes#140]
I think we should have the run_model function include some automatic checks on, at least, cross-chain convergence. 
Initial thoughts on options:
1. automatically summarize the parameters that fail some threshold Rhat value. .e.g, return a list of the parameter names and Rhat values, as well as throw an error that says "n parameters failed to converge across chains, check your model results carefully and consider a longer burn-in".
2. automatically extend the model run for another n_iterations and re-sample the posterior, if the number of parameters that fail the Rhat test is >~ %5 of the monitored parameters.
3. add the extended model run as an option in the run_model function (e.g., auto_converge = TRUE)

**Comment by AdamCSmithCWS** [comment]
first suggestion is now integrated into the run-model function.
second is still a good idea
third might be dangerous (extremely long model runs for 1 or 2 extreme parameters)

**Comment by BrandonEdwards** [comment]
Just circling back on some outstanding issues. I think for (3), it might make sense to include an auto_converge = TRUE, but have an upper limit for the number of re-runs. We could set that by default, and/or allow the user to set it. I'm happy to take on that task, I'll assign myself here.

**Comment by AdamCSmithCWS** [comment]
We should also incorporate the [posterior](https://mc-stan.org/posterior/articles/posterior.html#summaries-and-diagnostics-1) package, which estimates an updated version of Rhat. The updated Rhat is much less sensitive to asymmetrical posteriors, such as those that are common for the n-values in these models. 



Vehtari A., Gelman A., Simpson D., Carpenter B., & Bürkner P. C. (2020). Rank-normalization, folding, and localization: An improved Rhat for assessing convergence of MCMC. Bayesian Analysis.

**Comment by BrandonEdwards** [comment]
@AdamCSmithCWS Good idea re posterior package! Are you planning on adding that in on your end and submitting a PR, or would you like me to cover that? Doesn't matter to me!

Also, with that updated way of calculating Rhat, does it now make it less "dangerous" to have that 3rd option be implemented, where the model continues to run until fully converged (again,perhaps up until a certain number of total iterations)?
